From b2c6c3265fcdc2a7614958a3af17b4826fe12307 Mon Sep 17 00:00:00 2001
From: Leah Rowe <leah@libreboot.org>
Date: Fri, 7 May 2021 19:39:34 +0100
Subject: [PATCH 5/6] asus/kgpe-d16: Allow loading Option ROMs (so that we can
 use graphics cards)

Since 49d99fcebc7c3860def51812c2f723ad3e8a4ea3 Option ROMs can't be loaded
PCI graphics cards anymore when "Graphics initialization" is set to "Use
native graphics init". This means only onboard graphics can be used until
the OS or SeaBIOS initialise PCI graphics cards.
This patch allows us to do a little hack. We can set "Use native graphics
init" to "Run VGA Option ROMs" and activate "Load Option ROMs on PCI
devices". This way coreboot will load the Option ROM on PCI graphics cards
which means payloads like GRUB can display graphics.
There's only one catch: onboard graphics won't work anymore. Because of this
we'll include seavgabios.bin (compiled by SeaBIOS) in CBFS. This will give
us the option to use both PCI graphics cards and onboard graphics that work
with native graphics init.

This is copied from the same patch for KCMA-D8
---
 src/mainboard/asus/kgpe-d16/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/mainboard/asus/kgpe-d16/Kconfig b/src/mainboard/asus/kgpe-d16/Kconfig
index ffbfd53169..95016a0c7f 100644
--- a/src/mainboard/asus/kgpe-d16/Kconfig
+++ b/src/mainboard/asus/kgpe-d16/Kconfig
@@ -30,7 +30,6 @@ config BOARD_SPECIFIC_OPTIONS
 	select MAINBOARD_HAS_LPC_TPM
 	select DRIVERS_I2C_W83795
 	select DRIVERS_ASPEED_AST2050
-	select MAINBOARD_FORCE_NATIVE_VGA_INIT
 	select POWER_STATE_DEFAULT_ON_AFTER_FAILURE
 	select IPMI_KCS
 
-- 
2.25.1

